pkg/virt-launcher/virtwrap/access-credentials/access_credentials.go

```diff
@@ -597,3 +546,81 @@ func (l *AccessCredentialManager) HandleQemuAgentAccessCredentials(vmi *v1.Virtu
 
 	return nil
 }
+
+type accessCredentialsInfo struct {
+	// secret name mapped to authorized_keys in that secret
+	secretMap map[string][]string
+	// filepath mapped to secretNames
+	userSSHMap map[string][]string
+	// maps users to passwords
+	userPasswordMap map[string]string
+}
+
+func (a *accessCredentialsInfo) addAccessCredential(accessCred *v1.AccessCredential) error {
+	secretName := getSecret(accessCred)
+	if secretName == "" {
+		return nil
+	}
+
+	secretDir := getSecretDir(secretName)
+	files, err := os.ReadDir(secretDir)
+	if err != nil {
+		return fmt.Errorf("error encountered reading secrets file list from base directory %s: %w", secretDir, err)
+	}
+
+	if isSSHPublicKey(accessCred) {
+		for _, user := range accessCred.SSHPublicKey.PropagationMethod.QemuGuestAgent.Users {
+			a.userSSHMap[user] = append(a.userSSHMap[user], secretName)
+		}
+
+		var authorizedKeys []string
+		for _, file := range files {
+			if file.IsDir() || strings.HasPrefix(file.Name(), "..") {
+				continue
+			}
+
+			pubKeyBytes, err := os.ReadFile(filepath.Join(secretDir, file.Name()))
+			if err != nil {
+				return fmt.Errorf("error encountered readding access credential secret file [%s]: %w", filepath.Join(secretDir, file.Name()), err)
```

Comment:
Can you please change it to `error occurred while reading the access credential secret file [%s]: %w`?

Reply:
Done